hilti/toolchain/src/compiler/optimizer.cc

```diff
@@ -583,6 +583,26 @@ struct TypeVisitor : OptimizerVisitor {
         }
     }
 
+    void operator()(type::Name* n) final {
+        auto t = n->resolvedType();
+        assert(t);
```

Comment:
Nit: I wonder whether we actually _need_ the `assert` here and couldn't instead return early like in many other places in this file. An unresolved `type::Name` shouldn't be "used" if we never needed to resolve it; it's likely some internal inconsistency, but we do not need to abort for it here.

Reply:
At this point a name expression should never be left unresolved, the validator would have caught that and aborted. I was debating if we need the check at all, just felt better to have the assert just in case.